### PR TITLE
feat: Introduce buffer timeout

### PR DESCRIPTION
**Pull request by gbence**
`bufferTimeout` enables Trap to guarantee that events don't wait longer in the client buffer than it is expected.

### CODE REVIEW

File: src/trap.js [hunk 1]
```diff
@@ -130,6 +130,11 @@ class Trap {
     this._buffer.bufferSizeLimit = bufferSizeLimit;
   }
 
+  // `bufferSizeLimit` setter proxy
```
Comment (denessapi) [inline src/trap.js:133]: bufferTimeout

File: test/trap.buffer-timeout.test.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+  afterAll(() => {
+    // Disable fetch() mocks
+    disableFetchMocks();
+
+    // Use real timers from now on
+    jest.useRealTimers();
+  });
+
+  test('sends chunks automatically on buffer fill', () => {
```
Comment (denessapi) [inline test/trap.buffer-timeout.test.js:45]: on buffer timeout

File: src/trap.js (hunk 1 above)
Comment (denessapi) [inline src/trap.js:133]: bufferTimeout

**Review by denessapi — CHANGES_REQUESTED** [review]
Just minor things

File: src/trap.js (hunk 1 above)
Comment (gbence) [reply src/trap.js]: Done

File: test/trap.buffer-timeout.test.js (hunk 2 above)
Comment (gbence) [reply test/trap.buffer-timeout.test.js]: Done

**Pull request merged**